apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -903,44 +903,37 @@ export class InstallManager {
           // Example: "C:\MyRepo\common\temp\npm-local\node_modules\.bin\npm"
           const packageManagerFilename: string = this._rushConfiguration.packageManagerToolFilename;
 
-          const packageManagerEnv: NodeJS.ProcessEnv = process.env;
-
-          if (
-            this._rushConfiguration.packageManagerOptions &&
-            this._rushConfiguration.packageManagerOptions.environmentVariables
-          ) {
-            for (const envVar in this._rushConfiguration.packageManagerOptions.environmentVariables) {
-              if ({}.hasOwnProperty.call(this._rushConfiguration.packageManagerOptions.environmentVariables, envVar)) {
-                let setEnvironmentVariable: boolean = true;
-                const envVarValueInRushConfig: string =
-                  this._rushConfiguration.packageManagerOptions.environmentVariables[envVar].value;
-
-                console.log(`\nProcessing definition for environment variable: ${envVar}`);
-
-                if (process.env[envVar]) {
-                  setEnvironmentVariable = false;
-                  console.log(colors.yellow(`WARNING: Environment variable already defined on the device:`));
-                  console.log(`  Name: ${envVar}`);
-                  console.log(`  Value set on the device: ${process.env[envVar]}`);
-                  console.log(`  Value set in Rush config: ${envVarValueInRushConfig}`);
-                  if (
-                    this._rushConfiguration.packageManagerOptions.environmentVariables[envVar].override &&
-                    this._rushConfiguration.packageManagerOptions.environmentVariables[envVar].override === true
-                  ) {
-                    setEnvironmentVariable = true;
-                    console.log(colors.yellow(`WARNING: Overriding the environment variable with the value set in Rush config.`));
-                  } else {
-                    console.log(colors.yellow(`WARNING: Not overriding the value of the environment variable.`));
-                  }
-                }
+          let packageManagerEnv: NodeJS.ProcessEnv = process.env;
 
-                if (setEnvironmentVariable === true) {
-                  console.log(`Setting environment variable for package manager.`);
-                  console.log(`  Name: ${envVar}`);
-                  console.log(`  Value: ${envVarValueInRushConfig}`);
-                  packageManagerEnv[envVar] = envVarValueInRushConfig;
-                }
-              }
+          if (this._rushConfiguration.packageManager === 'npm') {
+            if (
+              this._rushConfiguration.npmOptions &&
+              this._rushConfiguration.npmOptions.environmentVariables
+            ) {
+              packageManagerEnv = InstallManager._mergeEnvironmentVariables(
```

Comment:
Instead of calling this three times, stick it in a temporary variable and call it once.